### DISCUSSION TITLE
Find out when best fitness was achieved

### DISCUSSION THREAD

**Discussion by felix-knopp** [ultralytics/yolov5#6308] [Q&A] [answered]
Is there a way to see in which epoch best fitness was achieved? 
After a training i want to know in which epoch i got best.pt. 

I could use the results.csv to calculate fitness for each epoch and look for the maximum.
Is there an easier way?

Maybe it would be nice to include fitness in results.csv aswell :)

**Comment by glenn-jocher** [comment, marked as answer]
@felix-knopp fitness is heavily weighted towards mAP@0.5:0.95 by default, so just looking for the max point there would be a good proxy. The fitness function is here:
https://github.com/ultralytics/yolov5/blob/436ffc417ac2312de18287ddc4f87bdc2f7f5734/utils/metrics.py#L15-L19

**Reply by glenn-jocher** [reply]
@felix-knopp BTW you can also track your trainings with W&B to see everything plotted out, i.e.:
https://wandb.ai/glenn-jocher/YOLOv5_v6.0_official

**Reply by felix-knopp** [reply]
@glenn-jocher thanks! I am already using W&B. It's a wonderful tool!
I thought maybe the epoch in which best.pt was achieved is logged somewhere. I guess not.
I'll just write a script that gets max fitness from results.csv then.

**Reply by glenn-jocher** [reply]
@felix-knopp no it's not saved anywhere currently other than the EarlyStopping instance. If EarlyStopping occurs then the best epoch is displayed to screen on training end, see https://github.com/ultralytics/yolov5/pull/5303

```python
            LOGGER.info(f'Stopping training early as no improvement observed in last {self.patience} epochs. '
                        f'Best results observed at epoch {self.best_epoch}, best model saved as best.pt.\n'
                        f'To update EarlyStopping(patience={self.patience}) pass a new patience value, '
                        f'i.e. `python train.py --patience 300` or use `--patience 0` to disable EarlyStopping.')
```
